api/net/Areas/Subscriber/Controllers/ContentController.cs

```diff
@@ -106,10 +111,7 @@ public class ContentController : ControllerBase
     public async Task<IActionResult> FindWithElasticsearchAsync([FromBody] JsonDocument filter, [FromQuery] bool includeUnpublishedContent = false)
     {
         // Exclude any content the user is not allowed to search for.
-        var username = User.GetUsername() ?? throw new NotAuthorizedException("User is missing");
-        var user = _userService.FindByUsername(username) ?? throw new NotAuthorizedException($"User [{username}] does not exist");
-        user = _userService.FindById(user.Id) ?? throw new NotAuthorizedException($"User [{username}] does not exist");
-
+        var user = _impersonate.GetCurrentUser();
```

Comment:
Nearly every time we check for the current user we now will check if they are impersonating.